ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -162,16 +228,26 @@ queryDecodeNodeToClient codecConfig queryVersion blockVersion
         blockVersion
       return (SomeSecond (BlockQuery blockQuery))
 
-instance SerialiseResult blk (BlockQuery blk) => SerialiseResult blk (Query blk) where
+instance  ( SerialiseResult blk (BlockQuery blk)
+          , Serialise (HeaderHash blk)
+          ) => SerialiseResult blk (Query blk) where
   encodeResult codecConfig blockVersion (BlockQuery blockQuery) result
     = encodeResult codecConfig blockVersion blockQuery result
   encodeResult _ _ GetSystemStart result
     = toCBOR result
+  encodeResult _ _ GetChainBlockNo result
+    = toCBOR result
+  encodeResult _ _ GetChainPoint result
+    = encodePoint encode result
 
   decodeResult codecConfig blockVersion (BlockQuery query)
     = decodeResult codecConfig blockVersion query
   decodeResult _ _ GetSystemStart
     = fromCBOR
+  decodeResult _ _ GetChainBlockNo
+    = fromCBOR
+  decodeResult _ _ GetChainPoint
+    = decodePoint decode
```

Comment:
I've inlined code that was previously in the typeclass.

Reply:
@nfrisby I've inlined the type class instance here.  Although it's unclear if that fixes the problem.

Reply:
I think it does, at least for me. Because now, if we need version-dependent serialization of `Point` in the future, then we can scrutinize the in-scope `version` argument here (currently named `_`).